integration_tests/e2e/overview/preInductionOverview.cy.ts

```diff
@@ -46,7 +46,7 @@ context('Prisoner Overview page - Pre Induction', () => {
   it('should render prisoner Overview page listing no prisoner goals given prisoner has no goals created pre-induction', () => {
     // Given
     const prisonNumberForPrisonerWithNoGoals = 'A00001A'
-    cy.task('getActionPlan', prisonNumberForPrisonerWithNoGoals)
+    cy.task('getGoalsByStatus404', prisonNumberForPrisonerWithNoGoals)
```

Comment:
I'm not sure about returning a 404 rather than a response that contains an empty array.
I think we need to think about and understand the semantics of that.

I think this test, arguably possibly badly written (I'll take that), might highlight the problem (because it didnt fail!)
Previously, an action plan with no goals returns a 200 response with an empty goals array. And the UI rendering would not display any goals:
```
{# Template to render the prisoner's list of Goals on the Overview tab #}

{% for goal in goals.goals %}
  <div class="govuk-summary-card" data-qa="goal-summary-card">
    <div class="govuk-summary-card__title-wrapper">
      <h3 class="govuk-summary-card__title">Goal {{ goal.sequenceNumber }}
....
```
IE. the collection is empty and the loop would not loop

And we assert that (perhaps badly) in this test with the `.hasNoGoalsDisplayed` method
```
  hasNoGoalsDisplayed(): OverviewPage {
    this.goalSummaryCards().should('not.exist')
    return this
  }
...
  private goalSummaryCards = (): PageElement => cy.get('[data-qa=goal-summary-card]')
```

By returning a 404, unless you are specifically catching and handling it in your service somewhere, then I _think_ `problemRetrievingData` will be `true` and you'll be rendering the "service is down" content:
```
<section data-qa="post-induction-overview">
  {% if not goals.problemRetrievingData %}

    {% include './_actionsHeader.njk' %}
    {% include '../_goalsList.njk' %}

  {% else %}
    <h2 class="govuk-heading-m">Sorry, the service is currently unavailable.</h2>
    <p class="govuk-body">
      Retrieving goals is currently unavailable. Please try again later.
    </p>
  {% endif %}
</section>
```
I think we need to think about the semantics of what "no goals" vs a 404 means 🤔

Reply:
Scrap that, I've just found the code in the service method that handles the 404:
```
    } catch (error) {
      if (error.status === 404) {
        logger.debug(`No plan created yet so no goals with [${status}] for Prisoner [${prisonNumber}]`)
        return { goals: undefined, problemRetrievingData: false }
```

Cool 👍

Reply:
The 404 is handled as not an error and initialise an empty array in the service so you don't get the service unavailable text. I made this test return a 404 as it's labeled as pre-induction so this should simulate the behaviour of the real service.